pkg/ocmextensions/cred.go

```diff
@@ -0,0 +1,93 @@
+package ocmextensions
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	authnK8s "github.com/google/go-containerregistry/pkg/authn/kubernetes"
+	"github.com/open-component-model/ocm/pkg/contexts/credentials"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var ErrNoAuthSecretFound = errors.New("no auth secret found")
+
+func GetAuthnKeychain(ctx context.Context,
+	credSecretSelector *metav1.LabelSelector,
+	clnt client.Client,
+) (authn.Keychain, error) {
+	secretList, err := getCredSecrets(ctx, credSecretSelector, clnt)
+	if err != nil {
+		return nil, err
+	}
+	return authnK8s.NewFromPullSecrets(ctx, secretList.Items)
+}
+
+func GetCredentials(ctx context.Context,
+	registryCredValue []byte,
+	ociRegistry *OCIRegistry,
+	clnt client.Client,
+) (credentials.Credentials, error) {
+	credSecretSelector, err := GenerateLabelSelector(registryCredValue)
+	if err != nil {
+		return nil, err
+	}
+	secretList, err := getCredSecrets(ctx, credSecretSelector, clnt)
+	if err != nil {
+		return nil, err
+	}
+	keychain, err := authnK8s.NewFromPullSecrets(ctx, secretList.Items)
```

Comment:
It looks like these lines could be replaced by `GetAuthnKeychain(...)` ?

Reply:
yes, replaced